### PR TITLE
Add template support for the `message` option

### LINKED ISSUE #176: Allow to define an update message in a template with arguments
I understand that I can pass the `message` argument to the `notify()` function. However I would like to specify a slightly different message but keep the original formatting. Right now I have to install `chalk`, import it, and construct own message. 
It would be nice to have to be able to specify `message` option as a template with variables that are processed internally without me having to import `chalk` and work with `notifier.update` variables.

An example:

```javascript
notifier.notify({
    message: `Update available {current} → {latest}
Run {command} to update.

My custom message.`
});
```

It's not really a big deal to format the message by myself but that would save some time.

### PR DESCRIPTION
**Pull request by fisker**
Make it easier to use custom message.

Fixes #176

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -154,7 +154,21 @@ Defer showing the notification to after the process has exited.
...
 Default: [See above screenshot](https://github.com/yeoman/update-notifier#update-notifier-)
 
-Message that will be shown when an update is available.
+Message that will be shown when an update is available, support [`pupa`](https://github.com/sindresorhus/pupa) template.
+
+Available placeholders:
+
+- `{name}` - Package name.
+- `{current}` - Current version.
+- `{latest}` - Latest version.
```
Comment (sindresorhus) [inline readme.md]: I think these should be `currentVersion` and `latestVersion`.

File: readme.md [hunk 2]
```diff
@@ -154,7 +154,21 @@ Defer showing the notification to after the process has exited.
 Type: `string`\
 Default: [See above screenshot](https://github.com/yeoman/update-notifier#update-notifier-)
 
-Message that will be shown when an update is available.
+Message that will be shown when an update is available, support [`pupa`](https://github.com/sindresorhus/pupa) template.
```
Comment (sindresorhus) [inline readme.md]: I don't think we need to mention `pupa`. Can just say that it supports placeholders that will be replaced.

File: readme.md [hunk 3]
```diff
@@ -154,7 +154,21 @@ Defer showing the notification to after the process has exited.
...
+
+Available placeholders:
+
+- `{name}` - Package name.
+- `{current}` - Current version.
+- `{latest}` - Latest version.
+- `{command}` - Install command.
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
- `{updateCommand}` - Update command.
```

File: readme.md [hunk 4]
```diff
@@ -154,7 +154,21 @@ Defer showing the notification to after the process has exited.
...
+
```
Comment (sindresorhus) [inline readme.md]: Should we also expose the update type? https://github.com/yeoman/update-notifier/blob/adf78037fbe4d33f06dd22a36db296902360b590/index.js#L109 We don't use it, but could be useful, I think.

File: readme.md [hunk 5]
```diff
@@ -154,7 +154,21 @@ Defer showing the notification to after the process has exited.
 Type: `string`\
 Default: [See above screenshot](https://github.com/yeoman/update-notifier#update-notifier-)
 
-Message that will be shown when an update is available.
+Message that will be shown when an update is available, support [`pupa`](https://github.com/sindresorhus/pupa) template.
+
+Available placeholders:
+
+- `{name}` - Package name.
```
Comment (sindresorhus) [inline readme.md]:
```suggestion
- `{packageName}` - Package name.
```

File: readme.md [hunk 6]
```diff
@@ -154,7 +154,21 @@ Defer showing the notification to after the process has exited.
...
+- `{current}` - Current version.
+- `{latest}` - Latest version.
+- `{command}` - Install command.
+
+```js
+notifier.notify({message: 'Run `npm install {name}@{latest}` to update.'});
```
Comment (sindresorhus) [inline readme.md]: This is not a good example as people should use the `{updateCommand}` placeholder so it works with Yarn too.

File: test/notify.js [hunk 7]
```diff
@@ -58,6 +58,44 @@ test('use pretty boxen message by default', t => {
 `);
 });
 
+test('support custom message', t => {
```
Comment (sindresorhus) [inline test/notify.js]:
```suggestion
test('supports custom message', t => {
```

File: test/notify.js [hunk 8]
```diff
@@ -58,6 +58,44 @@ test('use pretty boxen message by default', t => {
...
+	const notifier = new Control();
+	notifier.notify({
+		defer: false,
+		isGlobal: true,
+		message: 'custom message'
+	});
+
+	t.not(stripAnsi(errorLogs).indexOf('custom message'), -1);
```
Comment (sindresorhus) [inline test/notify.js]:
```suggestion
	t.true(stripAnsi(errorLogs).includes('custom message'));
```

File: readme.md (hunk 4 above)
Comment (fisker) [reply readme.md]: I've tried to add this, but seems this is from `options` passed by user, so maybe not useful.

**Pull request merged**